### PR TITLE
Create `signchecksum` script

### LINKED ISSUE #393: Add signed checksums to releases
It's a pity there's still no checksums to verify the integrity of the file downloaded, nor a gpg signature. It shouldn't be an "issue", but a normal process for a crypto software :-(

### PR DESCRIPTION
**Pull request by lukechilds**
Resolves #393 

So ideally we'd have checksumming and signing automated by our build system. However there's currently no simple way to do this without forking our build tools and adding this functionality or significantly changing our build setup.

For now I've just created a script that will pull down all build artefacts from the  latest GitHub release and create checksums and sign them. We can then manually upload the checksum file. I'm using my personal PGP key to sign the checksums. I think for now it's fine for any core developer to use this script to sign the checksums as their identity can be verified on https://keybase.io.

Going forward it would be great if we can work with `electron-builder` to integrate this functionality into their tooling. It must be a pretty common requirement. We should probably also create a dedicated HyperDEX release PGP key for signing.

**Not sure if this script really belongs in the code base but it might be good to have here until we have an automated solution in case we need to do a release when I'm not around.**

### CODE REVIEW

**Comment by lukechilds** [comment]
I've added checksums to the `v0.1.0-alpha.8` release, here's an example of what it looks like:

https://github.com/hyperdexapp/hyperdex/releases/download/v0.1.0-alpha.8/SHASUMS256.txt.asc

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

d70e55190569aa02ebd5215857d68723ca207e94c6df6677aa4979f8037e05a8  HyperDEX-0.1.0-alpha.8.dmg
817ab5a8b299ad81967942f2b90a185a71cc27dc8a8f788829579500a0f2191e  HyperDEX-0.1.0-alpha.8.dmg.blockmap
4eccdf95a528afbe326c9b4f83218c893bfc222c8c3dd16fb8fb7535a1fd70a3  hyperdex-0.1.0-alpha.8-mac.zip
1bdabc5ad5cf5037ada10d2e68a36603251c1d63f3031b0753e78982f7b7d747  hyperdex-0.1.0-alpha.8-x86_64.AppImage
5e3d253efc48b8bfc404134a9268b420619d6973991fc53198f3a8bf1b4afa1c  hyperdex-0.1.0-alpha.8.x86_64.rpm
e96fe44aa94651f11d67a3a864bbb75e2623d67a4037662903b55fba814624c5  hyperdex-setup-0.1.0-alpha.8.exe
ef70e37e2541c8172db04757f2b8bf1cf4a9b018e5d883b15ee8c3677d9870a7  hyperdex-setup-0.1.0-alpha.8.exe.blockmap
4cf802d600fecfa709468797362a2ac3bc92a03c24f5c170bc20db84c1f64e75  hyperdex_0.1.0-alpha.8_amd64.deb
0b7eea7f5e18ea19a99fb48f4b61134b114405a0711bfe1978965f1af3dd5f35  hyperdex_0.1.0-alpha.8_amd64.snap
3ff56111e9c562136b2b2723d8621cba40d83e5349e28ccfd11fa34130e74945  latest-linux.yml
8675997743de7b627df82564fd333cf34143729570405116c5139ef4d2d5b62b  latest-mac.yml
4219256ac797b41935c3352c4b8100260a5c19bbebe7140e0a21a2c934f1e698  latest.yml
-----BEGIN PGP SIGNATURE-----

iQEzBAEBCAAdFiEE6p8E5WGh0QmH4+Lmk9YSqdYcKBoFAls54cUACgkQk9YSqdYc
KBpPZQgAgxb5Ugz1Rlcz3Vandgv2LPAslHpSIAZ/SVc+b4rQcu7PXN/iIZzOyll4
bgeMEflFUt/QloQcuTirVKMUrJ+AIYPBeetQ7B0cLioR2MnZYMBIWLlne72Ocdhh
JHCYnnJpMzed8sFlTvUNutBUu/cVCp6Os3CGcxXGTxed1fjF1sSzC4lzj9/UAyew
eC3dn9DL4qcjfdMmRZFM0I5gobRexbfOeNnfvv7ZJqIxvKfR0KttsZYQvwgBqguC
iktOvnQZWLTZiLYfQKfA0KyfGgw9Buhy7XvKUS4Z4d7U3b8cpS9N+PXiAhpPW/jG
EEtHqOF4eXWoxCLxPYkrTYDLPATgjw==
=qfaI
-----END PGP SIGNATURE-----
```

File: signedchecksum [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+echo "Downloading build artefacts to temporary directory:"
+pwd
+
+curl https://api.github.com/repos/hyperdexapp/hyperdex/releases/latest | jq '.assets[].browser_download_url' | xargs wget
+
+echo "Calculating checksums and signing the result..."
+
+export GPG_TTY=$(tty)
+shasum -a 256 * | gpg --clearsign > "$run_dir/SHASUMS256.txt.asc"
```
Comment (sindresorhus) [inline signedchecksum]: Use `--algorithm` for readability.

**Comment by sindresorhus** [comment]
Can you also document it here https://github.com/hyperdexapp/hyperdex/blob/master/MAINTAINING.md#release ?

**Comment by iujzvbc** [comment]
me:$ gpg --verify SHASUMS256.txt.asc 
[cut]
gpg:                using RSA key EA9F04E561A1D10987E3E2E693D612A9D61C281A
gpg: Good signature from "Luke Childs <lukechilds123@gmail.com>" [expired]
gpg: Note: This key has expired!
Primary key fingerprint: EA9F 04E5 61A1 D109 87E3  E2E6 93D6 12A9 D61C 281A

**Comment by lukechilds** [comment]
You can verify with my renewed key at https://keybase.io/lukechilds

Looks like I forgot to update the pgp.mit.edu keyserver. I just pushed my latest key there, may take a while to update but you should be able to verify against my latest key in a few hours.

**Pull request merged**